### PR TITLE
[Mellanox] Enhancement for fan led management

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
**- What I did**

1. Add fan drawer class to Mellanox platform API
2. Refactor led code for Fan class and Psu class, put all led logic in led.py
3. Add device data to avoid sku specific code

**- How I did it**

**- How to verify it**

Manually test, regression is on going

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/led.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            if SharedLed.LED_PRIORITY[virtual_led.get_led_color()] < SharedLed.LED_PRIORITY[target_color]:
+                target_color = virtual_led.get_led_color()
+
+        return self._led.set_status(target_color)
+
+    def get_status(self):
+        return self._led.get_status()
+
+
+class VirtualLed(object):
```
Comment (Junchao-Mellanox) [inline platform/mellanox/mlnx-platform-api/sonic_platform/led.py:187]: Change name VirtualLed to ComponentFaultyIndicator.

File: platform/mellanox/mlnx-platform-api/sonic_platform/psu.py [hunk 2]
```diff
@@ -75,7 +68,9 @@ def __init__(self, psu_index, sku):
         else:
             filemap = psu_profile_list[0]
 
-        if sku in hwsku_dict_with_unplugable_psu:
+        self.psu_data = DEVICE_DATA[sku]['psus']
+
+        if not self.psu_data['hot_swappable']:
             self.always_presence = True
```
Comment (Junchao-Mellanox) [inline platform/mellanox/mlnx-platform-api/sonic_platform/psu.py:74]: change always_presence to always_present

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 3]
```diff
@@ -29,6 +28,7 @@
 MLNX_NUM_PSU = 2
 
 GET_HWSKU_CMD = "sonic-cfggen -d -v DEVICE_METADATA.localhost.hwsku"
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:30]: remove this one?

Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:30]: sku is still in use by other place,  i think we should remove this one when another PR is merged.

**Comment by Junchao-Mellanox** [comment]
Community PR https://github.com/Azure/sonic-buildimage/pull/4437

**Pull request closed without merging**